src/EnergyPlus/ZoneTempPredictorCorrector.cc

```diff
@@ -3608,7 +3608,10 @@ namespace ZoneTempPredictorCorrector {
                 if (TempCtrlFound && TempControlledZone(RelativeZoneNum).DeltaTCutSet > 0.0) {
                     TempZoneThermostatSetPoint(ZoneNum) = TempControlledZone(RelativeZoneNum).ZoneThermostatSetPointLo;
                     ZoneThermostatSetPointLo(ZoneNum) = TempControlledZone(RelativeZoneNum).ZoneThermostatSetPointLo;
-                    if (ZoneAirSolutionAlgo == Use3rdOrder) Tprev = ZTM1(ZoneNum);
+                    if (ZoneAirSolutionAlgo == Use3rdOrder) {
+                        Tprev = MAT(ZoneNum);
+                        if (ShortenTimeStepSys) Tprev = XMPT(ZoneNum);
+                    }
                     if (ZoneAirSolutionAlgo == UseAnalyticalSolution || ZoneAirSolutionAlgo == UseEulerMethod) Tprev = ZoneT1(ZoneNum);
```

Comment:
Also, all of this stuff related to onoff control should be skipped if there aren't any onoff thermostats in the idf.  As it stands now, this [loop to find RelativeZoneNumber](https://github.com/NREL/EnergyPlus/pull/6754/files#diff-26a720ee34299122c810df4c89b63101L3581) happens on every call, no matter what type of thermostat control these is or isn't. This whole RelativeZoneNum thing is unnecessary, but that's for another day. For now, mimic the staged thermostats and add `NumOnOffCtrZone` flag similar to `NumStageCtrZone ` that skips all of this if there aren't any.

Also, it seems that all of this (which was move here in #6499) would be better placed [here](https://github.com/NREL/EnergyPlus/blob/b60abaa44542b4495671647f18044141b63b1ac2/src/EnergyPlus/ZoneTempPredictorCorrector.cc#L3060) in `PredictSystemLoads` rather than `CalcPredictedSystemLoads`. This would make it similar to the staged thermostat and would put this before some of the zone air temp updates which might be a better fix than this one.

Reply:
@mjwitte The section was moved into PredictSystemLoads.